### ISSUE TITLE
reloading tests fails on chrome when using ember test --server

### ISSUE THREAD

**Issue by kenglxn** [ember-cli/ember-cli#5110]
I have a repo demonstrating the failure, including steps to reproduce, and workaround/fix: https://github.com/kenglxn/ember-cli-reload-bug

Basically when making changes to a test, or adding a new test, tests blow up in chrome due to integrity check with the following message:

`Uncaught Error: Assertion Failed: The tests file was not loaded. Make sure your tests index.html includes "assets/tests.js"`

![browser](https://raw.githubusercontent.com/kenglxn/ember-cli-reload-bug/master/browser.png "Browser screenshot")

![console](https://raw.githubusercontent.com/kenglxn/ember-cli-reload-bug/master/console.png "Console screenshot")

**Comment by kenglxn** [comment]
In phantomJS the failure is shown as expected, but in chrome it blows up.

**Comment by rwjblue** [comment]
This is an issue with ember-cli-sri.  I haven't had time to dig into what exactly is going on, but the fingerprint for tests.js isn't being updated properly when it is changed causing an error in Chrome (and none of the tests run).

Folks being bit by this and looking for a work around should use the changes in #5111 until we can get ember-cli-sri fixed.

**Comment by jonathanKingston** [comment]
So somewhere between `ember-cli@1.12.0` (the last time I had a fake app checked end to end) and `ember@1.13.10` something has changed causing the html file to be cached. Even a full browser refresh shows the stale HTML rather than the new hashes.

I'm not really sure #5111 would be enough as mine is failing on the app files too when they change. The better fix would either be to either:
- prevent hashing on this file completely
- prevent cached html
- prevent SRI to run in test mode

**Comment by rwjblue** [comment]
> prevent hashing on this file completely
> prevent SRI to run in test mode

My concern with these options is that we are basically never going to have an opportunity to confirm SRI is actually working.

**Comment by jonathanKingston** [comment]
@rwjblue I agree totally, just enumerating my thoughts as it's late here :grinning:

Trying to see where the server would be caching the HTML, will continue to dig.

**Comment by rwjblue** [comment]
Walkthrough of what is happening today:

1. initial build runs
2. tests/index.html is updated with the integrity values for the various included assets (including `assets/tests.js`)
3. User changes files in `tests/*` (also applies to `app/` changes but the walkthrough is focusing on tests)
4. `assets/tests.js` changes 
5. broccoli-sri-hash checks if `tests/index.html` in its input tree has changed, since it has not the cached value is used

---

*tldr;* the broccoli-filter cache for `tests/index.html` is not busted when `assets/tests.js` changes so the old value is used. broccoli-sri-hash needs to be updated to extend from broccoli-caching-writer (instead of broccoli-filter).

**Comment by eddiesholl** [comment]
I can narrow down the version range to breaking somewhere between ember-cli 1.13.8 and 1.13.10

**Comment by artburkart** [comment]
Everything referenced in this issue appears to be merged and closed, but I still can't run `ember test`. Is there a status update with `1.13.13` for this?

**Comment by ZackMattor** [comment]
@artburkart just fixed this by running `ember init` and updating package.json and bower.json to the packages suggested. (ignore the ember and ember-data versions if you are on > 2.x.x) then clearing out my `bower_components` and `node_modules` directory and running `npm install && bower install`
